traveler-ios-ui/TravelerKitUI/ViewControllers/AvailabilityViewController.swift

```diff
@@ -88,112 +145,84 @@ open class AvailabilityViewController: UIViewController {
             Traveler.fetchPasses(product: product, availability: availability, option: nil, delegate: self)
         }
     }
-}
 
-extension AvailabilityViewController: UITableViewDataSource {
-    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return datePickerCellVisible ? 2 : 1
-    }
-
-    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        if indexPath.row == 0 {
-            let cell = tableView.dequeueReusableCell(withIdentifier: dateCellIdentifier, for: indexPath) as! DateCell
-            cell.valueLabel.textColor = (availabilityError != nil ? true : false) ? UIColor.red : UIColor.darkText
-            cell.titleLabel.textColor = (availabilityError != nil ? true : false) ? UIColor.red : UIColor.darkText
-
-            switch (availabilityError) {
-            case .some(BookingError.badDate):
-                cell.valueLabel.text = "Unavailable"
-            case .some(BookingError.noDate):
-                cell.valueLabel.text = "Please Select"
-            default:
-                cell.valueLabel.text = selectedAvailability.flatMap({ DateFormatter.yearMonthDay.string(from: $0.date) })
-            }
-
-            return cell
-        } else {
-            let cell = tableView.dequeueReusableCell(withIdentifier: datePickerCellIdentifier, for: indexPath) as! DatePickerCell
-            cell.datePicker.minimumDate = Date()
-            cell.datePicker.date = selectedAvailability?.date ?? Date()
-            cell.delegate = self
-            return cell
-        }
+    private func setupPageController() {
+        pageController = UIPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
+        pageController?.dataSource = self
+        pageController?.delegate = self
+
+        addChild(pageController!)
+        pageController?.view.autoresizingMask = [.flexibleWidth,.flexibleHeight]
+        pageController?.view.frame = calendarContainer.bounds
+        calendarContainer.addSubview(pageController!.view)
+        pageController?.didMove(toParent: self)
+
+        let storyboard = UIStoryboard(name: "BookingItem", bundle: Bundle(for: AvailabilityViewController.self))
+        let initialVC = storyboard.instantiateViewController(withIdentifier: "calendarVC") as! AvailabilityCalendarViewController
```

Comment:
The point of relying heavily in storyboards is to show how the flow behaves visually. Doing this defeats that purpose. If you're going to put a container view inside a view controller, then do all your setup and building  in the storyboard not here. Examples of this: Add flights, booking item details, etc. It is of paramount importance that you follow convention. The goal is for the codebase to look like it was written by a single person.